### PR TITLE
Separate ArcMap and ArcGIS Pro code #642

### PR DESCRIPTION
**Pull request by csmoore**
We are just switching the head branch we are merging PR #644 into (branch: **Pro2.6** vs. dev).

Implements #642
Original PR: https://github.com/Esri/coordinate-conversion-addin-dotnet/pull/644 (we will close this one unmerged after this PR is merged)

Original PR #644 comments from @sumitzarkar :

We have implemented code for  following points:
 1. Removing: ArcMapAddinCoordinateConversion project/dll
 2. Removing:CoordinateConversionLibrary project/dll
 3. Refactoring the Pro project to remove dependencies on former CoordinateConversionLibrary
   a. remove view inheritance
  b. remove no-longer-needed classes (super/abstract classes)
  c. remove mediator/dispatcher (ex: convert to RelayCommands)
  d. use Pro styling for Dark Mode support
5. As part of this if you notice any hard-coded UI strings that should be converted to Resources (for I18n) please convert these to resources

### CODE REVIEW

**Comment by csmoore** [comment]
@sumitzarkar We have finished our review and testing, the only remaining issues to close and merge this PR is:

Combined duplicate Base Classes into one class/file:

1. TabBaseViewModel and ProTabBaseViewModel
2. OutputCoordinateViewModel and ProOutputCoordinateViewModel

[NEW] A couple of Dark Mode Form issues:

1. #647 
2. #648

**Comment by sumitzarkar** [comment]
@csmoore 

>>Combined duplicate Base Classes into one class/file:
TabBaseViewModel and ProTabBaseViewModel
OutputCoordinateViewModel and ProOutputCoordinateViewModel

We are working on this.

**Comment by csmoore** [comment]
@sumitzarkar -as we discussed I added [2 more dark mode issues to the list above](https://github.com/Esri/coordinate-conversion-addin-dotnet/pull/646#issuecomment-595203790)

File: source/CoordinateConversion/ProAppCoordConversionModule/Models/CoordinateConversionLibraryConfig.cs [hunk 1]
```diff
@@ -25,7 +26,7 @@
 
 namespace ProAppCoordConversionModule.Models
 {
-    public class CoordinateConversionLibraryConfig : NotificationObject
+    public class CoordinateConversionLibraryConfig : ViewModelBase
```
Comment (csmoore) [inline source/CoordinateConversion/ProAppCoordConversionModule/Models/CoordinateConversionLibraryConfig.cs:29]: We may have been a little unclear in our [original feedback](https://github.com/Esri/coordinate-conversion-addin-dotnet/pull/644#discussion_r386048612), [ViewModelBase](https://pro.arcgis.com/en/pro-app/sdk/api-reference/#topic10691.html) is used for view model classes and [PropertyChangedBase](https://pro.arcgis.com/en/pro-app/sdk/api-reference/#topic10646.html) is used for all other classes.

**Comment by csmoore** [comment]
@sumitzarkar We have finished our review and testing, the only remaining issues to close and merge this PR is:

Combined duplicate Base Classes into one class/file:

TabBaseViewModel and ProTabBaseViewModel
OutputCoordinateViewModel and ProOutputCoordinateViewModel
[NEW] A couple of Dark Mode Form issues:

#647
#648

**Comment by csmoore** [comment]
✔️  These issues have been tested and closed (Dark Mode Form issues) : #647 , #648

So only remaining issues are : 

1. Combined duplicate Base Classes into one class/file:
TabBaseViewModel and ProTabBaseViewModel
OutputCoordinateViewModel and ProOutputCoordinateViewModel
2. Change a few base class objects from `ViewModelBase` to `PropertyChangedBase`  - comment here:  https://github.com/Esri/coordinate-conversion-addin-dotnet/pull/646/commits/eff8118d3ece199b7831cf2ff861ef7427fec30c#r389223919

**Comment by csmoore** [comment]
[Successfully tested](https://github.com/Esri/coordinate-conversion-addin-dotnet/issues/642#issuecomment-601932387) - merging

**Pull request merged**